### ISSUE TITLE
Passing description to the "command()" method causes error being produced in runtime

### ISSUE THREAD

**Issue by slavafomin** [tj/commander.js#976]
Hello!

Thank you for this great library!

However, when command description is passed as a second argument to the `Command.command()` method it causes the following error in runtime `error: foo-bar(1) does not exist, try --help`.

This doesn't work:

```
  program
    .command('build', 'Builds the package')
    .action(buildCommand)
  ;
```

But this works:
```
  program
    .command('build')
    .description('Builds the package')
    .action(buildCommand)
  ;
```

This is extremely counter-intuitive, especially when you explore the API using IDE autocompletion. When I see the `description` argument of the method, my first thought it to pass it to it.

Having two methods of passing the description with one of them triggering some different behavior looks pretty insane from the API standpoint.

**Comment by shadowspawn** [comment]
It is a common cause of confusion.

**Comment by andares** [comment]
I have same problem, when I try to attach a desc text to command, It's not work.

`alias` method also can not work.

```reason
program
  |> command("cc <dir>"/*, "Compile pattern to docker compose config"*/)
  /* |> description("Compile pattern to docker compose config") */
  /* |> alias("cc") */
  |> option("-o, --output", "Output dir")
  |> action(Cmd_Cc.main);
```

*coding with reasonml now* 😹

**Comment by shadowspawn** [comment]
FYI @andares (and I think @slavafomin worked this out after getting caught out by the current behaviour).

For action based command handling you pass the description separately from the command:

```
program
    .command('build')
    .description('Builds the package')
    .action(buildCommand);
```

For git-style executables you pass the description along with the command:

```
program
    .command('build', 'Builds the package')
    .action(buildCommand);
```

**Comment by andares** [comment]
if set description with command() function:

```javascript
var Process = require("process");
var Commander = require("commander");

var program = Commander.version("0.1.0");

program.allowUnknownOption(false);

program.command("cc <dir>", "Compile pattern to docker compose config").option("-o, --output", "Output dir").action((function (param) {
        console.log("test");
        return /* () */0;
      }));

program.parse(Process.argv);
```

there has problem when run:

![图片](https://user-images.githubusercontent.com/7556505/59922565-a7d5f180-9463-11e9-8398-f4a49635ca81.png)

**Comment by shadowspawn** [comment]
@andares 
Do not put the description in the call to `.command`, just pass one parameter which is the command name.

I fixed my comment with examples where I had a mistake, hopefully clearer now! (Sorry.)
https://github.com/tj/commander.js/issues/976#issuecomment-503346600

**Comment by andares** [comment]
I got it, thanks 😄

**Comment by shadowspawn** [comment]
This issue will be resolved when v3.0.0 is released. Available now as a prerelease. See #1001

Reworked the README and JSDoc and TypeScript to hopefully make the two modes of operation clearer. Have not changed the runtime, but will close this issue and see how the changes help.

**Comment by shadowspawn** [comment]
Reworked the README and JSDoc and TypeScript to hopefully make the two modes of operation clearer. Have not changed the runtime, but will close this issue and see how the changes help.

Shipped in v3: https://github.com/tj/commander.js/releases/tag/v3.0.0
